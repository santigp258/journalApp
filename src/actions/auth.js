import { types } from "../types/types";


//return login action
export const login = (uid, displayName) => ({
  type: types.login,
  payload: {
    uid,
    displayName,
  },
});
