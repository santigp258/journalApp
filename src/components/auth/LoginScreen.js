import React from "react";
import Swal from "sweetalert2";
import { useDispatch, useSelector } from "react-redux";

import { Link, useHistory } from "react-router-dom";
import { startGoogleLogin, startLoginEmailPassword } from "../../actions/auth";

import validator from "validator";
//import { removeError, setError } from "../../actions/ui";
import { useForm } from "../../hooks/useForm";

export const LoginScreen = () => {
  //history for redirect
  const history = useHistory();

  //dispatch for value. React-redux hook
  const dispatch = useDispatch();

  /*   //delete other errors
  useEffect(() => {
    dispatch(removeError("No email"));
  }, [dispatch]); */

  //hook useForn
  const [formValues, handleInputChange] = useForm({
    email: "santiago123@gmail.com",
    password: "12345",
  });

  //content ui state //msgError
  const { loading } = useSelector((state) => state.ui);
  const { email, password } = formValues;

  const handleLogin = (e) => {
    e.preventDefault();

    //send action
    if (isFormValid()) {
      dispatch(startLoginEmailPassword(email, password));
      history.replace("/");
    }
  };

  //login with Google
  const handleGoogleLogin = () => {
    dispatch(startGoogleLogin());
  };

  const isFormValid = () => {
    if (!validator.isEmail(email)) {
      Swal.fire("Error", "Is not email. Try valid email", "error");
    } else if (password.trim().length < 5) {
      Swal.fire(
        "Error",
        "Password should be at least 6 characters and match each other",
        "error"
      );
      return false;
    } else {
      return true;
    }
  };

  return (
    <>
      <h3 className="auth__title">Login</h3>
      <form onSubmit={handleLogin}>
        {/*  {msgError && <div className="auth__alert-error">{msgError}</div>} */}
        <input
          className="auth__input"
          type="text"
          placeholder="Email"
          name="email"
          autoComplete="off"
          value={email}
          onChange={handleInputChange}
        />
        <input
          className="auth__input"
          type="password"
          placeholder="Password"
          name="password"
          value={password}
          onChange={handleInputChange}
        />
        <button
          type="submit"
          className="btn btn-primary btn-block"
          disabled={loading}
        >
          Login
        </button>
        <div className="auth__social-networks">
          <p>Login with Social Network</p>
          <div className="google-btn" onClick={handleGoogleLogin}>
            <div className="google-icon-wrapper">
              <img
                className="google-icon"
                src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
                alt="google button"
              />
            </div>
            <p className="btn-text">
              <b>Sign in with google</b>
            </p>
          </div>
        </div>

        <Link to="/auth/register" className="link">
          Create a new account
        </Link>
      </form>
    </>
  );
};
