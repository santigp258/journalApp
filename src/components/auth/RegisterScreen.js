import React, { useLayoutEffect } from "react";
import { Link } from "react-router-dom";
import { useForm } from "../../hooks/useForm";
import validator from "validator";
import { useDispatch, useSelector } from "react-redux";
import { removeError, setError } from "../../actions/ui";
import { startRegisterWithEmailPassword } from "../../actions/auth";

export const RegisterScreen = () => {
  //errors
  const dispatch = useDispatch(setError);

  //delete other errors
  useLayoutEffect(() => {
    dispatch(removeError("No email"));
  }, []);

  //content ui state
  const { msgError } = useSelector((state) => state.ui);

  const [formValue, handleInputChange] = useForm({
    name: "",
    email: "",
    password: "",
    password2: "",
  });

  const { name, email, password, password2 } = formValue;

  //register onsubmit
  const handleRegister = (e) => {
    e.preventDefault();
    if (isFormValid()) {
      console.log("Hi!");
      dispatch(startRegisterWithEmailPassword(email, password, name));
    }
  };

  const isFormValid = () => {
    if (name.trim().length === 0) {
      dispatch(setError("Name is required"));
      return false;
    } else if (!validator.isEmail(email)) {
      dispatch(setError("No email"));
    } else if (password !== password2 || password.length < 5) {
      dispatch(
        setError(
          "Password should be at least 6 characters and match each other"
        )
      );
      return false;
    } else {
      return true;
    }
  };
  return (
    <>
      <h3 className="auth__title">Register</h3>
      <form onSubmit={handleRegister}>
        {msgError && <div className="auth__alert-error">{msgError}</div>}
        <input
          className="auth__input"
          type="text"
          placeholder="Name"
          name="name"
          onChange={handleInputChange}
          autoComplete="off"
        />
        <input
          className="auth__input"
          type="text"
          placeholder="Email"
          name="email"
          onChange={handleInputChange}
          autoComplete="off"
        />
        <input
          className="auth__input"
          type="password"
          onChange={handleInputChange}
          placeholder="Password"
          name="password"
        />
        <input
          className="auth__input"
          type="password"
          placeholder="Confirm password"
          name="password2"
          onChange={handleInputChange}
        />
        <button type="submit" className="btn btn-primary btn-block mb-5">
          Register
        </button>

        <Link to="/auth/login" className="link">
          Already Registered?
        </Link>
      </form>
    </>
  );
};
