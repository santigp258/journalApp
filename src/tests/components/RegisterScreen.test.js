import { mount } from "enzyme";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router";
import { RegisterScreen } from "../../components/auth/RegisterScreen";

import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

const middlewares = [thunk];
//create store
const mockStore = configureStore(middlewares);

const initState = {
  auth: {},
  ui: { loading: false, msgError: null },
};

let store = mockStore(initState);

const wrapper = mount(
  <Provider store={store}>
    <MemoryRouter>
      <RegisterScreen />
    </MemoryRouter>
  </Provider>
);
describe("pruebas en <RegisterScreen/>", () => {
  test("debe de mostarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
