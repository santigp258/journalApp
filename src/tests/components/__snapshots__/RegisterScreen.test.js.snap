// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <RegisterScreen/> debe de mostarse correctamente 1`] = `
Array [
  <h3
    className="auth__title"
  >
    Register
  </h3>,
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth__input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
    />
    <input
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
    />
    <input
      className="auth__input"
      name="password2"
      onChange={[Function]}
      placeholder="Confirm password"
      type="password"
    />
    <button
      className="btn btn-primary btn-block mb-5"
      type="submit"
    >
      Register
    </button>
    <a
      className="link"
      href="/auth/login"
      onClick={[Function]}
    >
      Already Registered?
    </a>
  </form>,
]
`;
